### DISCUSSION TITLE
Can StreetComplete be used for adding/confirming new POIs?

### DISCUSSION THREAD

**Discussion by NKAmapper** [streetcomplete/StreetComplete#5390] [Q&A]
**Use case**

I have a number of files with POIs (nodes) which would be useful in OSM, for example all kindergartens in Norway. High quality data, geocoded, tagged and import plan already reviewed. But each node would benefit from verification by a human + slight adjustment of coordinates to get a sensible location for the node at the given address + possibly minor edit of the name tag.
It can be done in MapRoulette, but StreetComplete has a much higher usage in my community. Data can be provided in a geojson file.
[Import plan](https://wiki.openstreetmap.org/wiki/Key:no-barnehage:nsrid)

**Proposed Solution**

Is this type of task supported by StreetComplete?

**Comment by westnordost** [comment]
Short answer: No.

Long answer: Theoretically yes, but something along these lines would need to be implemented first. Since there are possibly many people with a list of POIs from a third party data source (that is license compatible), it would make sense to have some sort of backend in which these could be managed so that on StreetComplete-side, there would just be *one* quest that talks with *one* API to get a list of to-be-confirmed new POI places.

Maybe MapRoulette could be this backend. I don't know (what it can do, and if there is an API for that).

However, a few additional notes here:
- matching with data already present in OSM is going to be non-trivial. (I.e. no quest should be displayed for a POI that already exists in OSM - at a maybe slightly different position. Maybe it was added very recently.)
- if confirming that the POI really exists requires being on-site, then one could instead also map all the POIs with the "shop overlay" anyway. Sure, this quest would make it so that people do not have to have that overlay active, but still...

(Also, converting this to a discussion, because this is a question)

**Reply by NKAmapper** [reply]
Agree that it would be great to have one setup to cater for all POI-type imports.

In the case of kindergartens I would be able to filter out most of the approx. 1500 kindergartens which are already in OSM, however I am sure some would stille have to be resolved manually, and that would ideally be part of the quest.

Some types of POIs (for example kindergartens) may usually be mapped from the armchair (outdoor area visible on aerials), others would benefit from on-site survey.

**Reply by matkoniecz** [reply]
https://github.com/alltheplaces/alltheplaces seems highly relevant and a potential aggregated source

I have write-up of what needs to be implemented to work nicely in OSM, and applied for funding so I have time to implement this. I am now waiting for reply (note that it is not highly likely that I will get funding from this specific source)

I am also waiting for LWG to oficially publish their opinion about what kind of info from website of shop/shop chain we can use for OSM mapping.

> Some types of POIs (for example kindergartens) may usually be mapped from the armchair

not sure about kindergartens, though for say fuel stations it may work

**Reply by westnordost** [reply]
Oh, nice... 

Does the data have to be ODBL licence compatible if the data is only used as a hint to lead on-site mappers to the place to confirm that it is really there? After all, when the mapper is on-site, the `source` of the information added to OSM is `survey` after all. (Not including phone numbers, websites and what else might have been scraped for alltheplaces)

Though I can understand if it is a problem because it could be interpreted as a thin veil over a copyright infringement.

**Reply by westnordost** [reply]
Also, regarding alltheplaces:

Does this also cover this specific use case though? I.e. someone comes around with a big dataset from a third party that is basically a static file in geojson format or something? From the readme, it sounds more like it is a collection of scrapers.

**Reply by matkoniecz** [reply]
I prefer to wait for official minutes, but from what I understand: database rights are not triggered for shop or chain of shops listing their shops with locations/opening hours/names/available payment modes etc. So scrapping like ATP is OK if it is directly from shop pages.

But scrapping from Google Maps or OSM triggers database rights.

warning: not a lawyer, I could misunderstood something, I am also waiting for LWG to publish minutes before doing anything like this

**Reply by matkoniecz** [reply]
> Does the data have to be ODBL licence compatible if the data is only used as a hint to lead on-site mappers to the place to confirm that it is really there?

no, it is fine (if I understand right, I am not a lawyer) to use any source to plan places to visit and then map purely from survey

(though, arriving at location confirming that shop with a given name exists and adding shop with its type or opening hours copied from some source without checking actual shop type or its opening hours would not count as "map purely from survey")

-------------------------------------------------------------

but it would be nice to offer way to quickly add entry based on available info (for example, with opening hours already filled and shop type already selected and name already added, with user only confirming existence and possibly improving location info)

**Reply by westnordost** [reply]
There is a missing puzzle piece though. There must be an API in-between alltheplaces and STreetComplete. An API that likely has to record false-positives so that not every mapper is asked again and again if some non-existing shop exists at a location. (see also #5199)

**Reply by NKAmapper** [reply]
Just to be clear, the POIs I had in mind are all from sources where the owner (government or corporate) have given explicit permission. We have many data sets like this but lack an easy to use app like StreetComplete to get more users engaged in easy verifications and adjustments.

**Reply by matkoniecz** [reply]
> There is a missing puzzle piece though.

Yes, that one would need to run as part of SC infrastructure (like photo service) or be part of ATP.

Another missing part is deciding whether given dataset is good enough to be proposed (suggested making this part of ATP, can be partially manual and partially based on false positive ratio).

**Reply by matkoniecz** [reply]
> have given explicit permission

That drops "we need to wait for LWG to publish minutes" part but otherwise implementation effort and interface would be the same (or greater if that is not aggregated at ATP).

> We have many data sets like this but lack an easy to use app like StreetComplete to get more users engaged in easy verifications and adjustments.

Rapid kind of tries this, but it is armchair mapping with all consequences.

**Reply by matkoniecz** [reply]
And to be clear, I would be really happy to work on it and use it and have it available in StreetComplete but it is much larger project that say street furniture overlay that can be added in some chunks of hobby time.

**Comment by mnalis** [comment]
Related discussions/issues about using external data as quest source in StreetComplete:
- https://github.com/streetcomplete/StreetComplete/issues/5199
- https://github.com/streetcomplete/StreetComplete/discussions/2477
- https://github.com/streetcomplete/StreetComplete/discussions/4019
- https://github.com/streetcomplete/StreetComplete/issues/3738
- https://github.com/streetcomplete/StreetComplete/issues/5389 (which created this discussion, but GitHub does not link discussions neatly)

Also, [SCEE](https://wiki.openstreetmap.org/wiki/SCEE) fork of StreetComplete has functionality to create quest from (externally generated) `.csv` file which is related tunable functionality, but it is local-device only (and not distributed among different users, which is wanted in many cases like here).
- https://github.com/Helium314/SCEE/issues/362


Perhaps having some server on which such datasets can be setup, and SC/SCEE functionality to "subscribe" to wanted datasets (assuming there will be more than one dataset available in a country)? In that case, server would be relatively simple (e.g. see what data SCEE `.csv` functionality has: basically `lat`, `lon`, `text`, `node tags`) but solved quests could be removed from everybody lists thus eliminating wasted duplicate effort. That is assuming  functionality similar to what SCEE has for confirming objects from `.csv` file is satisfactory.

The main issue there (but with other solutions too) IMHO would be how to detect duplicates (probably should detect main tag that the quest wants to add e.g. `amenity`, `leisure`, `building` and show all of those around current location?)
